api/pkg/transformer/server/server.go

```diff
@@ -17,42 +21,87 @@ type Options struct {
 
 // Server serves various HTTP endpoints of Feast transformer.
 type Server struct {
-	options *Options
+	options    *Options
+	httpClient *http.Client
+	logger     *zap.Logger
 
-	PreprocessHandler  func(w http.ResponseWriter, r *http.Request) ([]byte, error)
-	PostprocessHandler func(w http.ResponseWriter, r *http.Request) ([]byte, error)
+	PreprocessHandler  func(ctx context.Context, request []byte) ([]byte, error)
+	PostprocessHandler func(ctx context.Context, request []byte) ([]byte, error)
 	LivenessHandler    func(w http.ResponseWriter, r *http.Request)
 }
 
-// NewStandardTransformer initializes a new Server.
-func New(o *Options) *Server {
+// New initializes a new Server.
+func New(o *Options, logger *zap.Logger) *Server {
 	return &Server{
-		options: o,
+		options:    o,
+		httpClient: &http.Client{},
+		logger:     logger,
 	}
 }
 
+// PredictHandler handles prediction request to the transformer and model.
 func (s *Server) PredictHandler(w http.ResponseWriter, r *http.Request) {
-	var out []byte
-	var err error
+	ctx := r.Context()
+
+	request, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		s.logger.Error("read request body", zap.Error(err))
+		fmt.Fprintf(w, err.Error())
+		return
+	}
+	defer r.Body.Close()
+	s.logger.Debug("request", zap.ByteString("request", request))
+
+	response := request
 
 	if s.PreprocessHandler != nil {
-		out, err = s.PreprocessHandler(w, r)
+		response, err = s.PreprocessHandler(ctx, request)
 		if err != nil {
+			s.logger.Error("preprocess error", zap.Error(err))
 			fmt.Fprintf(w, err.Error())
 			return
 		}
-		log.Println("Preprocess output:", string(out))
+		s.logger.Debug("preprocess response", zap.ByteString("preprocess_response", response))
 	}
 
-	// TODO: call the model predict url
+	response, err = s.predict(ctx, response)
+	if err != nil {
+		s.logger.Error("predict error", zap.Error(err))
+		fmt.Fprintf(w, err.Error())
+		return
+	}
+	s.logger.Debug("predict response", zap.ByteString("predict_response", response))
+
+	// TODO: Postprocessing (next milestone)
+
+	fmt.Fprintf(w, string(response))
+}
 
-	// TODO: call postprocesshandler
+func (s *Server) predict(ctx context.Context, request []byte) ([]byte, error) {
+	req, err := http.NewRequest("POST", s.options.ModelPredictURL, bytes.NewBuffer(request))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
I think you should pass the headers from the original request. Because if logging enabled, original request to transformer will has requestID headers generated which we need to pass to model prediction request